kedro/framework/cli/pipeline.py

```diff
@@ -549,10 +576,13 @@ def _install_files(  # pylint: disable=too-many-locals
     )
 
     if conf_source.is_dir() and alias:
-        alias_name = alias.split(".")[-1]
-        _rename_files(conf_source, package_name, alias_name)
+        _rename_files(conf_source, package_name, alias)
+
+    if destination:
+        module_path = f"{destination}.{alias or package_name}"
+    else:
+        module_path = f"{alias or package_name}"
```

Comment:
And in fact, we can make this easier on ourselves and make `module_path` be an actual `Path` object. Then this code becomes `module_path = Path(alias or package_name)` etc. 

Maybe we can also turn `destination` into a `Path` as quickly as possible, rather than pass it around as a string until the very end.